src/edu/stanford/nlp/sempre/FloatingParser.java

```diff
@@ -100,14 +176,12 @@ protected int getBeamSize() {
     return Parser.opts.beamSize;
   }
 
-
-
   // Construct state names.
   private Object floatingCell(String cat, int depth) {
-    return cat + ":" + depth;
+    return (cat + ":" + depth).intern();
```

Comment:
Does interning actually help?  In my experience, it always seemed like a good idea, but doesn't really improve speed.